scipy/stats/_hypotests.py

```diff
@@ -389,3 +391,214 @@ def _get_wilcoxon_distr(n):
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided'
+          * 'less': one-sided
+          * 'greater': one-sided
+
+    pooled : bool, optional
+        Whether to compute score statistic with pooled variance or
+        unpooled (Wald statistic)
+
+    num_it : int, optional
+        Number of iterations of the grid search. Default is 4. Must be
+        non-negative.
+
+    Returns
+    -------
+    ber : ``BarnardExactResult`` object
+            The object has attributes ``score_statistic`` and ``p_value``.
+            ``score_statistic`` is the Z statistic with pooled (Score) or
+            unpooled (Wald) variance, depending on the user choice of
+            `pooled` param.
+            ``p_value`` the probability of obtaining a test statistic at least
+            as extreme as the one that was actually observed, assuming that
+            the null hypothesis is true.
+
+    See Also
+    --------
+    chi2_contingency : Chi-square test of independence of variables in a
+        contingency table.
+    fisher_exact : Fisher exact test on a 2x2 contingency table.
+
+    Notes
+    -----
+    Barnard's test is an exact test used in the analysis of contingency
+    tables. It examines the association of two categorical variables and
+    is a more powerful alternative than Fisher's exact test
+    for 2x2 contingency tables
+
+    References
+    ----------
+    .. [1] `Mehta, Cyrus & Senchaudhuri, Pralay. (2003). Conditional
+    versus Unconditional Exact Tests for Comparing Two Binomials`__.
+
+    .. __: https://www.statsols.com/hubfs/Resources_/Comparing-Two-Binomials.pdf
+
+    Examples
+    --------
+    examples from [1]_.
+
+    Consider the following example of a vaccine efficacy study (Chan, 1998).
+    In a randomized clinical trial of 30 subjects, 15 were innoculated with a
+    recombinant DNA influenza vaccine and the 15 were innoculated with a
+    placebo. Twelve of the 15 subjects in the placebo group (80%) eventually
+    became infected with influenza whereas for the vaccine group, only 7 of
+    the 15 subjects (47%) became infected. The data are tabulated as a 2 × 2
+    table::
+
+            Vaccine  Placebo
+        Yes     7        12
+        No      8        3
+
+    We use this table to find the p-value:
+    >>> import scipy.stats as stats
+    >>> res = stats.barnard_exact([[7, 12], [8, 3]], alternative="less")
+    >>> res.score_statistic
+    1.894...
+    >>> res.pvalue
+    0.03407...
+
+    In this exemple, we use the `alternative` parameter with the "less" option,
+    because the vaccine has either no effect (H0 hypothesis) or a 'positive'
+    effect (H1 hypothesis). Fisher’s exact test produces an exact
+    p-value of 0.0641 (with
```

Comment:
What is the thought - or is there precedent - for choosing the meanings of 'less' and 'greater' here?
Is there a way of stating this that sounds intuitive (e.g. 'less' to me sounds like 'negative', 'greater' sounds like 'positive') rather than the opposite of intuition?
Should the meanings of 'greater' and 'less' be more precisely defined in the notes, as in gh-13188, for example, to avoid confusion?

Reply:
Since in the initial issue, the author of the issue was talking of `fisher_exact` test, I decided to start from this test to have a similar behavior, but maybe it wasn't the best idea. In [R](https://cran.r-project.org/web/packages/Barnard/Barnard.pdf), barnard_exact return two p_value : `two_sided` and `one_sided`.

So, in order to reproduce `fisher_exact` behavior, I tried to interpret what was the meaning of less, greater and two sided, which is explained in [wikipedia](https://en.wikipedia.org/wiki/One-_and_two-tailed_tests). 
But maybe I was wrong, and there was a sense in `fisher_exact` test and there isn't in `barnard_exact` test. As a side note, there isn't any mention on less or greater alternatives in the paper I linked in the documentation. 

What is your opinion @mdhaber ?

Reply:
In Boschloo's test (which I'd like to implement when `barnard_exact` will be merged into master), there is a definition of the less and greater alternative tests (you can find it here : https://en.wikipedia.org/wiki/Boschloo%27s_test#Test_hypothesis). I guess it is the same in `barnard_exact`.
Would you like me to add more explanations into the `barnard_exact`'s doc?

Reply:
> Would you like me to add more explanations into the barnard_exact's doc?

Yes, I think that would be a good idea.